src/coord/Weather.cc

```diff
@@ -38,8 +39,13 @@ Weather::Weather(double airTemperature, double airPressure, double humidity)
 }
 
 bool Weather::operator==(Weather const& other) const noexcept {
-    return (_airTemperature == other.getAirTemperature() && _airPressure == other.getAirPressure() &&
-            _humidity == other.getHumidity());
+    bool tempMatch = (std::isnan(_airTemperature) && std::isnan(other.getAirTemperature())) ||
+                     _airTemperature == other.getAirTemperature();
+    bool presMatch = (std::isnan(_airPressure) && std::isnan(other.getAirPressure())) ||
+                     _airPressure == other.getAirPressure();
+    bool humiMatch =
+            (std::isnan(_humidity) && std::isnan(other.getHumidity())) || _humidity == other.getHumidity();
+    return (tempMatch && presMatch && humiMatch);
```

Comment:
I have the same concerns about NaN equality here as in my comment at https://github.com/lsst/geom/pull/35#pullrequestreview-795905964.

Reply:
Here I think there's a much stronger case, because `Weather`, `Observatory`, and `VisitInfo` cannot even be thought of as mathematical objects or number-like in any way.

In these objects, NaNs are used unofficially to mean "unknown" or "not applicable". This is especially true for `VisitInfo`, where the pybind11 wrapper allows for any field to be undefined by default. This whole rabbit-hole started because two `VisitInfo` objects did not compare equal because a _single field_ was set to NaN in both.